Update fluent-bit image to 2.0.4

iamleot opened the pull request:
Closes #274.

Signed-off-by: Leonardo Taccari <leonardo@faire.ai>

edsiper commented:
FYI: we are shipping 2.0.4 right now

iamleot commented:
> FYI: we are shipping 2.0.4 right now

Whoops, I've missed that!

If no candidate PR is prepared for the Helm chart I can adjust this one to update to `2.0.4` instead!

Thank you @edsiper!

stevehipwell commented:
@edsiper is there anything which would need to change to support v2? Is YAML GA yet (if not where can I feedback on it as the use of arrays is problematic)?

I'm still planning on splitting the charts out into collector and aggregator variants so major work can happen then.

edsiper commented:
- 2.0.4 notes: https://fluentbit.io/announcements/v2.0.4/
- no breaking changes
- YAML -> I would encourage more testing

braunsonm commented:
@stevehipwell YAML is still problematic with arrays, see https://github.com/fluent/fluent-bit/issues/5446
Workaround works there.

When YAML goes GA it would be good to have better support in the chart for it (currently you need to override the start params and use a custom config file)

iamleot commented:
Moving back to draft... rebasing and pushing with the force to update this PR to 2.0.4!

stevehipwell commented on charts/fluent-bit/Chart.yaml:
```diff
@@ -5,8 +5,8 @@ keywords:
   - fluentd
-version: 0.20.10
-appVersion: 1.9.9
+version: 0.20.11
```
```suggestion
version: 0.21.0
```
I think this probably warrants a bigger version bump.

iamleot replied:
Sure!

stevehipwell reviewed (changes requested):
Thanks for the PR @iamleot, just one comment for you to take a look at.

stevehipwell commented:
@iamleot I'm going to try and merge a couple of other PRs before yours as the current automation is very noisy and it'd be best to get the noise out of the way before the v2 chart release. You might need to rebase once they're done but I'll comment back here when you can do it without needing to do so multiple times.

stevehipwell commented:
Just waiting for https://github.com/fluent/helm-charts/pull/278.

stevehipwell commented:
@iamleot could you rebase this now?

stevehipwell commented:
@iamleot do you want to remove the draft status?

iamleot commented:
> @iamleot do you want to remove the draft status?

Yeah, I was waiting for the GitHub Actions workflows checks to complete to avoid possible further notification spamming!

I have rebased to latest `main` and hopefully addressed all the comments!

If there is anything else please let me know!

Thank you very much!

stevehipwell reviewed (approved):
LGTM
